### PR TITLE
added_test_for_ShoppingCart1

### PR DESCRIPTION
**Pull request by OlegShvets1**

### CODE REVIEW

**Comment by OlegShvets1** [comment]
Screenshot_1
Screenshot_2

File: src/test/resources/database/books/delete-user-from-db.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+DELETE FROM users;
```
Comment (okuzan) [inline src/test/resources/database/books/delete-user-from-db.sql]: add an empty line to every script, where you see this symbol
image

File: src/test/java/mate/academy/bookstore/service/ShoppingCartServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        when(authentication.getPrincipal()).thenReturn(user);
+        when(cartItemRepository.findById(cartItemId)).thenReturn(Optional.of(cartItem2));
+        when(shoppingCartRepository.findByUserId(userId)).thenReturn(Optional.of(shoppingCart));
+        when(cartItemRepository.findByShoppingCartId(shoppingCart.getId()))
+                .thenReturn(Set.of(cartItem1, cartItem2));
+
+        shoppingCartService.deleteBookFromTheShoppingCart(authentication, cartItemId);
+
+        verify(shoppingCartRepository).findByUserId(userId);
+        verify(cartItemRepository).findById(cartItemId);
```
Comment (okuzan) [inline src/test/java/mate/academy/bookstore/service/ShoppingCartServiceTest.java:193]: why no assert of calling the delete method?

File: src/test/java/mate/academy/bookstore/controller/ShoppingCartControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        mockMvc.perform(get("/api/cart")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .with(authentication(authentication)))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").exists())
+                .andExpect(jsonPath("$.cartItems.length()").value(AMOUNT_OF_BOOKS));
+    }
+
+    @Sql(scripts = "classpath:database/books/add-books-to-shoppingCart-to-db.sql",
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
```
Comment (okuzan) [inline src/test/java/mate/academy/bookstore/controller/ShoppingCartControllerTest.java]: extract to class level, you have it on each method

File: src/test/java/mate/academy/bookstore/controller/ShoppingCartControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    @Sql(scripts = "classpath:database/books/delete-books-and-shoppingCart-from-db.sql",
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
```
Comment (okuzan) [inline src/test/java/mate/academy/bookstore/controller/ShoppingCartControllerTest.java]: same

**Review by okuzan — CHANGES_REQUESTED** [review]
Review your sql scripts annotation, remove redundancy, if scripts are always used together - merge them, if it should be applied to each method of the class - extract it to a test level

**Review by okuzan — APPROVED** [review]

**Pull request merged**